### PR TITLE
Update build dependencies

### PR DESCRIPTION
**Pull request by larsgrefer**
- update Gradle to 3.1
- update Realm to 2.0.0
- update Android to 24

### CODE REVIEW

**Comment by kneth** [comment]
@larsgrefer The build failure looks like a CI issue. The license for Android SDK Build-Tools 24.0.1 is not accepted.

@emanuelez Is the above something you can solve for us?

**Comment by emanuelez** [comment]
It looks like this file does not have a Dockerfile or a Jenkinsfile so I'm gonna fix that, which will make fixing this PR trivial.

File: build.gradle [hunk 1]
```diff
@@ -1,21 +1,20 @@
 apply plugin: 'ch.netzwerg.release'
-project.ext.sdkVersion = 23
-project.ext.buildTools = '23.0.0'
-project.ext.realmVersion = '1.0.1'
+project.ext.sdkVersion = 24
+project.ext.buildTools = '24.0.1'
+project.ext.realmVersion = '2.0.0'
```
Comment (dalinaum) [inline build.gradle]: I think you can update Realm to 2.0.1 now.

**Comment by emanuelez** [comment]
@larsgrefer Please merge the updated master branch to your PR. I implemented some changes that should make updating the build infrastructure easier. In the process, I already updated some of the libraries you updated as well. I tried not to, but I was running into some build errors that forced me to. Anyway, I didn't update Realm itself so I'll let that to you if you want.

Thank you for prompting this change! It was long needed :)

File: build.gradle (hunk 1 above)
Comment (larsgrefer) [inline build.gradle]: Done

File: build.gradle [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+        classpath 'com.android.tools.build:gradle:2.2.0'                    // Android Build tools
         classpath 'ch.netzwerg:gradle-release-plugin:1.2.0'                 // Manage Release tags / versioning
-        classpath 'com.jakewharton.sdkmanager:gradle-plugin:0.12.0'         // Manage Android SDK dependencies
-        classpath 'com.github.dcendents:android-maven-gradle-plugin:1.3'    // Allows you to install AAR's locally
+        classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'    // Allows you to install AAR's locally
         classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:4.0.1'   // ???
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.6'      // Bintray/OJO integration
         classpath 'com.novoda:gradle-android-command-plugin:1.5.0'          // ADB commands from Gradle (monkey)
-        classpath 'io.realm:realm-gradle-plugin:1.0.1'                      // Realm base library
+        classpath 'io.realm:realm-gradle-plugin:2.0.0'                      // Realm base library
```
Comment (larsgrefer) [inline build.gradle]: Done

**Comment by larsgrefer** [comment]
@emanuelez Seems like Jenkins is still unhappy

**Comment by emanuelez** [comment]
@larsgrefer 

```
:adapters:compileDebugAndroidTestJavaWithJavacNote: Processing class AllJavaTypes
Note: Processing module RealmTestModule
/tmp/realm-android-adapters/adapters/src/androidTest/java/io/realm/entity/UnsupportedCollection.java:37: error: UnsupportedCollection is not abstract and does not override abstract method isManaged() in RealmCollection
public class UnsupportedCollection<E extends RealmObject> implements OrderedRealmCollection<E> {
       ^
```

**Comment by emanuelez** [comment]
To make sure everything works locally, from the repo folder:
 * `docker build -t realm-adapters:latest .`
 * `docker run -ti -v $(pwd):/tmp realm-adapters:latest bash`
 * `cd tmp && ./gradlew javadoc check connectedCheck`

**Comment by emanuelez** [comment]
👍 The results of `android-adapters-pr` can be ignored now

**Comment by zaki50** [comment]
:+1:

**Comment by zaki50** [comment]
Thank you @larsgrefer 
I'll squash and merge this.

**Pull request merged**